### ISSUE TITLE
After building the index, occasionally the number of vectors increases.

### ISSUE THREAD

**Issue by Bennu-Li** [milvus-io/milvus#2768]
**Describe the bug**
milvus0.10.0
I inserted 10000000 64-dimensional vectors and then built the index. Then the number of vectors becomes 20000000.
This problem only happened once
Metadata changes when building index:
![image](https://user-images.githubusercontent.com/53458891/86770781-d3824480-c083-11ea-897f-bb853a4b628c.png)

![image](https://user-images.githubusercontent.com/53458891/86770856-eeed4f80-c083-11ea-9713-31e64a7a7d5b.png)

**Comment by yhmo** [comment]
Use this script to try reproduce:

```
# #!/usr/bin/env python

import sys, getopt
import struct
import time
import random
import string
import os

import json

import numpy as np
from milvus import Milvus, IndexType, Status
from milvus.client.abstract import CollectionSchema, IndexParam
from milvus.client.types import MetricType

SERVER_ADDR = "127.0.0.1"
SERVER_PORT = '19530'
TABLE_NAME = "AAA"
TABLE_DIMENSION = 64
TABLE_PARAM = {'collection_name': TABLE_NAME, 'dimension': TABLE_DIMENSION, 'index_file_size': 2048, 'metric_type':MetricType.L2}
INDEX_TYPE = IndexType.IVF_SQ8H
INDEX_PARAM = {'m': 16, 'nlist': 1024}

MILVUS = Milvus(host=SERVER_ADDR, port=SERVER_PORT)

def gen_vec_list(nb, seed=np.random.RandomState(1234)):
    xb = seed.rand(nb, TABLE_DIMENSION).astype("float32")
    vec_list = xb.tolist()
    return vec_list


def main(argv):
    for p in range(100):
        status = MILVUS.create_collection(TABLE_PARAM)
        total_start = time.time()
        vec_list = gen_vec_list(100000, seed=np.random.RandomState(1))
        for k in range(1):
            print("k=",k)
            for i in range(100):
                print("i=", i)
                status, ids = MILVUS.insert(collection_name=TABLE_NAME, records=vec_list)

        status = MILVUS.create_index(TABLE_NAME, index_type=INDEX_TYPE, params=INDEX_PARAM)

        total_end = time.time()
        print("Total cost:", total_end - total_start, 'sec')

        status, row_count = MILVUS.count_entities(collection_name=TABLE_NAME)
        print(TABLE_NAME, "row count =", row_count)
        if row_count != 10000000:
            print('ERROR!')
            break

        status = MILVUS.drop_collection(collection_name=TABLE_NAME)
        time.sleep(5)

if __name__ == "__main__":
    main(sys.argv[1:])

```

**Comment by yhmo** [comment]
The root reason: segment merge thread and build index thread overlapped

**Comment by yhmo** [comment]
Fixed by #2809

**Comment by yhmo** [comment]
Use the above script to test. 
Before fix: reproduce the bug at No.12 times. 
After fix: cannot reproduce after running 100 times.
